### PR TITLE
A try at iter_parse method (issue #29)

### PR DESCRIPTION
**Pull request by thomasballinger**
Concerns:
* trying regex iteratively is probably a big speed hit
* what to call the (ucs, is_sequence, capname, params) namedtuple
* duplicated logic in measure_length and identify_fragment
* what capnames to give catchalls at https://github.com/jquast/blessed/pull/77/files#diff-1b763dad61b521a08b7559c0363d2ff2R360

I haven't written tests or docs yet, I'm just checking for thoughts. Only tested with Python 3.5.

### CODE REVIEW

File: blessed/sequences.py [hunk 1]
```diff
@@ -40,7 +44,9 @@ def _sort_sequences(regex_seqlist):
     # types, it is feasible.
     # pylint: disable=bad-builtin
     #         Used builtin function 'filter'
-    return sorted(list(filter(None, regex_seqlist)), key=len, reverse=True)
+    for el in regex_seqlist:
+        assert isinstance(el, tuple)
```
Comment (thomasballinger) [inline blessed/sequences.py]: whoops

File: blessed/sequences.py [hunk 2]
```diff
@@ -96,12 +102,21 @@ def _build_any_numeric_capability(term, cap, num=99, nparams=1):
...
         cap_re = re.sub(r'(\d+)', r'(\d+)', cap_re)
         if r'(\d+)' in cap_re:
-            return cap_re
+            return cap, cap_re
         warnings.warn('Missing numerics in {0!r}: {1!r}'
                       .format(cap, cap_re), UserWarning)
     return None  # no such capability
 
 
+def _build_simple_capability(term, cap):
```
Comment (thomasballinger) [inline blessed/sequences.py]: use same language as other _build_* methods

**Comment by jquast** [comment]
Can we push this directly to ~jquast/blessed?

(I didn't think I had a preference, but TeamCity's service hook requires it. The build VCS settings are hardcoded to this repository, so I don't see how I could otherwise accomidate)

**Comment by thomasballinger** [comment]
Sure, and then open a PR from that to trigger the TeamCity service?

**Comment by jquast** [comment]
Yes, a new one please thankyou!

File: blessed/sequences.py [hunk 3]
```diff
@@ -816,3 +820,73 @@ def horizontal_distance(ucs, term):
...
+                outp += ucs[idx]
+                nxt += 1
+            else:
+                if outp:
+                    yield TextFragment(outp, False, None, None)
+                    outp = u''
+                yield identify_fragment(term, ucs[idx:idx+l])
+                nxt += l
+    if outp:
+        yield TextFragment(outp, False, None, None)
```
Comment (jquast) [inline blessed/sequences.py]: The hope is by reducing to a common function, we can find ways to optimize that function, feel free to innovate

**Pull request closed without merging**